blaze/compute/sql.py

```diff
@@ -585,6 +592,17 @@ def compute_up(expr, data, **kwargs):
     return table_of_engine(data, expr._name)
 
 
+@dispatch(DateTime, (ClauseElement, sa.sql.elements.ColumnElement))
+def compute_up(expr, data, **kwargs):
+    return getattr(sa.func, expr.attr)(data).label(expr._name)
```

Comment:
Select -> Select -> Table via .froms attribute
Table -> engine -> via .bind attribute
engine -> dialect via .dialect attribute

Or there might be a more direct way.  I've saved many an hour by poring over the list of attributes and private attributes on sqlalchemy expressions